tests/test_physics.py

```diff
@@ -434,3 +434,102 @@ def test_raycast():
             sim.set_stage_is_collidable(False)
             raycast_results = sim.cast_ray(test_ray_1)
             assert not raycast_results.has_hits()
+
+
+@pytest.mark.skipif(
+    not osp.exists("data/scene_datasets/habitat-test-scenes/apartment_1.glb"),
+    reason="Requires the habitat-test-scenes",
+)
+def test_collision_groups():
+    cfg_settings = examples.settings.default_sim_settings.copy()
+
+    # configure some settings in case defaults change
+    cfg_settings["scene"] = "data/scene_datasets/habitat-test-scenes/apartment_1.glb"
+
+    # enable the physics simulator
+    cfg_settings["enable_physics"] = True
+
+    # loading the physical scene
+    hab_cfg = examples.settings.make_cfg(cfg_settings)
+
+    with habitat_sim.Simulator(hab_cfg) as sim:
+        # get the rigid object attributes manager, which manages
+        # templates used to create objects
+        obj_template_mgr = sim.get_object_template_manager()
+        # get the rigid object manager, which provides direct
+        # access to objects
+        rigid_obj_mgr = sim.get_rigid_object_manager()
+
+        if (
+            sim.get_physics_simulation_library()
+            != habitat_sim.physics.PhysicsSimulationLibrary.NoPhysics
+        ):
+            cgh = habitat_sim.physics.CollisionGroupHelper
+            cg = habitat_sim.physics.CollisionGroups
+
+            # test group naming
+            assert cgh.get_group_name(cg.UserGroup1) == "UserGroup1"
+            assert cgh.get_group("UserGroup1") == cg.UserGroup1
+            cgh.set_group_name(cg.UserGroup1, "my_custom_group_1")
```

Comment:
Nit: sometimes we use unit tests to demonstrate functionality, almost like a tutorial. It would be nice if you showed a real robotics use case here with meaningful group names. As-is, users will probably wonder, why bother renaming enum UserGroup1 to "my_custom_group_1"? What am I supposed to do with this?

Reply:
Good point. This may make more sense once the AOs are merged in and we can do a grasping example and/or the override parameters to raycast/contactTest allow custom queries for things like sensor region objects.